packages/core-transaction-pool-mem/lib/connection.js

```diff
@@ -246,8 +266,7 @@ class TransactionPool extends TransactionPoolInterface {
         fetchSize,
       )) {
         const transaction = this.mem.getTransactionById(id)
-
-        if (transaction && this.checkApplyToBlockchain(transaction)) {
+        if (transaction) {
```

Comment:
If this is removed and I think transaction will always be defined, then the current function `getTransactionsForForging()` can be simplified to `this.getTransactionIdsForForging(0, blockSize).map(id => this.mem.getTransactionById(id).serialized)`

Reply:
Something like the last commit, still testing on forger

Reply:
changes something, but can be played later on...

Reply:
> the current function `getTransactionsForForging()` can be simplified to `this.getTransactionIdsForForging(0, blockSize).map(id => this.mem.getTransactionById(id).serialized)`

I mean that the entire body of the function can become 1 line.